libraries/libc/stdio/sscanf.cpp

```diff
@@ -0,0 +1,528 @@
+#include <assert.h>
+#include <ctype.h>
+#include <libutils/Scanner.h>
+#include <stdio.h>
+#include <string.h>
+
+namespace
+{
+enum
+{
+    SCANF_TYPE_CHAR,
+    SCANF_TYPE_SHORT,
+    SCANF_TYPE_INTMAX,
+    SCANF_TYPE_L,
+    SCANF_TYPE_LL,
+    SCANF_TYPE_PTRDIFF,
+    SCANF_TYPE_SIZE_T,
+    SCANF_TYPE_INT
+};
+}
+
+static void store_int(void *dest, unsigned int size, unsigned long long i)
```

Comment:
What about templates?

Reply:
Hm, how would you use templates here? The type is decided way before the invocation

Reply:
Using templates in libc breaks compatibility with C

Reply:
This code is part of the backend, we can use template or other C++ features if we want